Shopify/cli
# Run the dev command as service

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/620



### WHAT is this pull request doing?
- Don't prompt for selecting an organization when an app key is provided
  - Use a new shopify env variable
  - Other alternatives:
    - This could be avoided if we prompt for the organization only in case not `app key` is provided. In that case, the cli could get `orgId` after fetching the remote app information from the app key. That alternative make really complex the logic that fetches the dev data from the flags.

- Set the backend port to the one required by the spin systemctl
  - Use an existing spin env variable
  -  Other alternatives:
     - Now the CLI supports setting up the port in the `toml` config file. We could modify that file to add the port in the spin init script

-  Force the updating of the urls
   - When you run an app using the CLI in spin, the url is always to be the same. In order to skip this interactive prompt, we could assume that the url should be always updated. 
   - Other alternatives:
     - Configuring the final url of the 1P app in the spin init script. Then use the flag `--no-update` to skip that step
     - Adding a new flag that supports different updates modes `--update-mode always|yes|never|no` to skip the interactive prompt

- Don't display preview footer it the terminal does not support it
  - The cli check if the terminal used to run the dev command supports input raw mode. Otherwise only de preview url is displayed
  - Other alternatives:
     - Add a new flag or env variable to set the non-interactive mode. This way, regardless the terminal type, we could assign a default behaviour to every interactive prompt used along the run of the dev command
  
- Add env variables to inject a raw identity token
  - In case these new variables are set, the complete auth flow sets the session tokens from these values. During the rest of the execution, tokens stored in the session are always used. 


### How to test your changes?
- Run the `dev` command to check that the interactive mode is working properly
- You can follow the 1P app as a service [demo](https://shopify.workplace.com/groups/1207479046824983/permalink/1262249938014560/) to test these changes. That app includes some cli patches with these modifications.


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
